src/InteractionHandler.php

```diff
@@ -4,76 +4,144 @@
 
 namespace Exan\Fenrir;
 
+use Closure;
 use Exan\Fenrir\Component\Button\InteractionButton;
 use Exan\Fenrir\Constants\Events;
 use Exan\Fenrir\Enums\Parts\InteractionTypes;
 use Exan\Fenrir\Interaction\ButtonInteraction;
 use Exan\Fenrir\Interaction\CommandInteraction;
 use Exan\Fenrir\Parts\ApplicationCommand;
 use Exan\Fenrir\Rest\Helpers\Command\CommandBuilder;
+use Exan\Fenrir\Rest\Helpers\Command\QueuedCommand;
 use Exan\Fenrir\Websocket\Events\InteractionCreate;
 use Exan\Fenrir\Websocket\Events\Ready;
+use Throwable;
 
 class InteractionHandler
 {
     private FilteredEventEmitter $commandListener;
     private FilteredEventEmitter $buttonListener;
+    private string $applicationId;
+    private bool $devMode = false;
 
     /** @var array<string, callable> */
     private array $handlersCommand = [];
 
     /** @var array<string, callable> */
     private array $handlersButton = [];
 
-    private bool $devMode = false;
+    /** @var QueuedCommand[] */
+    private array $commandQueue = [];
+
+    /** @var string[] */
+    private array $guildsToCheck = [];
 
     public function __construct(private Discord $discord, private ?string $devGuildId = null)
     {
         if (!is_null($this->devGuildId)) {
             $this->devMode = true;
         }
+
+        $this->discord->gateway->events->once(
+            Events::READY,
+            function (Ready $ready) {
+                $this->applicationId = $ready->user->id;
+
+                $compareCommands = function (array $commands, bool $globalCommands) {
+                    foreach ($this->commandQueue as $queuedCommand) {
+                        if ($globalCommands && ($queuedCommand->guildId !== null)) {
+                            return;
+                        }
+
+                        $needsRegistered = true;
+
+                        foreach ($commands as $command) {
+                            if ($queuedCommand->commandBuilder->matchesApplicationCommand($command) === true) {
+                                $needsRegistered = false;
+                                break;
+                            }
+                        }
+
+                        if (!$needsRegistered) {
+                            continue;
+                        }
```

Comment:
This should be separated to remove the need for the storage of the `bool`

Something akin to:
```php
if (!$this->needsRegistration($queuedCommand, $commands)) {
    continue;
}
```

I think it'd also be cleaner to extract the functionality of this PR to a separate class entirely

```php
abstract class CommandRegistrar
{
    public function queueCommand(QueuedCommand $queuedCommand): Promise;

    public function registerQueued(string $applicationId): Promise; // Will resolve the promises of the queuedCommand returned above, and when each one is registered, resolve the Promise returned by itself
}

Reply:
I don't think this should be abstract. I do like...
```php
if (!$this->needsRegistration($queuedCommand, $commands)) {
    continue;
}
```
this suggestion but not so sure about extracting the comparing logic to their own class. Now splitting into dedicated methods I can understand.